airflow/policies.py

```diff
@@ -117,6 +117,19 @@ def get_dagbag_import_timeout(dag_file_path: str) -> int | float:  # type: ignor
     """
 
 
+@local_settings_hookspec
+def connection_policy(conn_id: str) -> str:  # type: ignore[empty-body]
+    """
+    Alter and/or apply validation rules on the provided connection id.
+
+    When used in combination with the task context (``airflow.operators.python.get_current_context``), you can
+    apply logic, e.g. to check if the requested connection id was requested by a DAG with a certain owner.
+
+    To support connection id alterations at runtime (typically prefixes are added), a connection id must be
+    returned, even when not altering the connection id.
+    """
```

Comment:
I am not sure, but I also wonder whether we should add a disclaimer that this could provide a false sense of security?  To me it makes sense to have this because people want _something_, and this is about as good as we can do right now.  But, it's pretty easy to circumvent e.g. with `secrets_backend_list[0].get_connection`.